### PR TITLE
List item component

### PR DESCRIPTION
**Pull request by AntonPDC**
So i accidentally added and committed to the input branch, but pushed on this new listItemComponent branch. I am not going to delete the inputcomponent branch until I know its all good.

Then I made another code cleanup push so there was that.

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import "./App.css";
-import { TodoList } from "./TodoList";
+import { TodoList } from "./TodoListMain";
```
Comment (ZacharyRizer) [inline src/App.tsx:2]: Let's change the name of this file and the component in the file to `Dashboard`.

File: src/TodoListMain.tsx [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import { useState } from "react";
```
Comment (ZacharyRizer) [inline src/TodoListMain.tsx:1]: This is the file that should be updated according the last comment.

File: src/TodoListMain.tsx [hunk 3]
```diff
@@ -1,6 +1,7 @@
...
 import Button from "./components/Button/Button";
 import Input from "./components/Input/Input";
+import TodoItemList from "./components/TodoItemList/TodoItemList";
 
 interface Todo {
```
Comment (ZacharyRizer) [inline src/TodoListMain.tsx:6]: So this interface already exists in your new `TodoItemList.tsx` file. Let's figure out how to not repeat code by exporting this from that file and importing it in this file along with the `TodoItemList`.

File: src/components/TodoItemList/TodoItemList.tsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  text: string;
+  completed: boolean;
+}
+
+interface TodoItemListProps {
+  todos: Todo[];
+  handleListToggle: (id: number) => void;
+}
+
+const TodoItemList: React.FC<TodoItemListProps> = ({
```
Comment (ZacharyRizer) [inline src/components/TodoItemList/TodoItemList.tsx:14]: Keep your style of code consistent in all files. Look at the `Button.tsx` and `Input.tsx` files and mimic the style for how the function is defined and how the props are defined.

Comment (AntonPDC) [reply src/components/TodoItemList/TodoItemList.tsx:14]: i believe i restructured the code to be similar

File: src/TodoListMain.tsx (hunk 3 above)
Comment (AntonPDC) [reply src/TodoListMain.tsx:6]: Made this harder than it had to be but I exported and imported it

File: src/components/TodoItemList/TodoItemList.tsx [hunk 5]
```diff
@@ -0,0 +1,33 @@
+import "./TodoItemList.css";
```
Comment (ZacharyRizer) [inline src/components/TodoItemList/TodoItemList.tsx:1]: Let's simplify the name of this component, its css file, and the directory it is in. Change all filenames and instances of `TodoItemList` to just `TodoList`.

File: src/Dashboard.tsx [hunk 6]
```diff
@@ -1,14 +1,10 @@
 import { useState } from "react";
 import Button from "./components/Button/Button";
 import Input from "./components/Input/Input";
+import { Todo } from "./components/TodoItemList/TodoItemList";
```
Comment (ZacharyRizer) [inline src/Dashboard.tsx:4]: You don't need two lines to import multiple things from the same file, condense this to a single line. Look up importing default and non-default imports in typescript.

File: src/Dashboard.tsx [hunk 7]
```diff
@@ -1,14 +1,10 @@
...
+import TodoItems from "./components/TodoItemList/TodoItemList";
 
-interface Todo {
-  id: number;
-  text: string;
-  completed: boolean;
-}
-
-export const TodoList: React.FC = () => {
+export const Dashboard: React.FC = () => {
```
Comment (ZacharyRizer) [inline src/Dashboard.tsx:7]: Convert this to the standard we've used in the other files: 

`function component_name() {}` with `export default compnent_name` at the bottom of the file.

File: src/components/TodoItemList/TodoItemList.tsx [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+
```
Comment (ZacharyRizer) [inline src/components/TodoItemList/TodoItemList.tsx:2]: You've changed the component name to `TodoItems`, but the filename and interface are still showing `TodoItemList....` Change the filenames (both tsx and css), the interface for the props, and the component to `TodoList`.

File: src/Dashboard.tsx (hunk 6 above)
Comment (AntonPDC) [reply src/Dashboard.tsx:4]: default imports go first. now i know this

**Review by ZacharyRizer — APPROVED** [review]

**Pull request merged**